data/scripts.js

```diff
@@ -247,7 +247,11 @@ let BattleScripts = {
 			if (damage === this.NOT_FAIL) pokemon.moveThisTurnResult = null;
 			if (damage || damage === 0 || damage === undefined) moveResult = true;
 		} else {
-			if (!targets.length) {
+			let notarget = !targets.length;
+			if (!notarget && this.gameType === 'triples' && ['adjacentFoe', 'adjacentAlly', 'normal', 'randomNormal'].includes(move.target)) {
+				notarget = !this.isAdjacent(targets[0], pokemon);
```

Comment:
My concern here is: How could this possibly be true?

Reply:
`targets` is assigned from

```js
		const {targets, pressureTargets} = pokemon.getMoveTargets(move, target);
```

This should not return a non-adjacent target if the move target-type requires adjacent.

Reply:
If the selected target has already fainted by the time the move executes then `getMoveTarget` calls `resolveTarget`, which tries to find a non-fainted adjacent foe. However, if this also fails then `resolveTarget` just returns any foe, preferring one that is not fainted. I tried changing that behaviour but it just made the tests hang so I went back to how it was done before the spread move refactor.

Reply:
It sounds like we should investigate why the change in `resolveTarget` makes tests hang.

Reply:
I've changed it now so that `resolveTarget` always returns an adjacent foe for normal targeting, preferring a non-fainted one, otherwise just using your opposite foe. This seems to fix Triples without breaking anything else.

Reply:
This is an improvement, but we should probably investigate why we can't return `null` here.

Reply:
#8735 for a [discussion](https://github.com/smogon/pokemon-showdown/pull/8735#discussion_r857179601) on ``null``